### PR TITLE
[WIP] Add security option in yum module

### LINKED ISSUE #11498: Add support for yum options related to security
#### ISSUE TYPE
Feature Idea

#### COMPONENT NAME
yum module

#### SUMMARY
yum these days comes with the ability to perform the following security related actions:

```
yum update --security # updates only the packages that need security related updates
yum update --cve CVE-2015-4000 # updates all packages with fixes for the given CVE
yum update --advisory=ALAS-2015-551 # updates packages in the scope of the given security advisory
```

As a person who administers fleets of servers, I often want to perform fixes that are security driven, and the above operations come in handy.

Ansible, however, has no support for these as of now.

Here's a page explaining these features: https://access.redhat.com/solutions/10021

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Fix adds option to specify security updates in yum module

Fixes #11498

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/packaging/os/yum.py

##### ANSIBLE VERSION
```
2.4 devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
This PR is in work in progress. I would love to have comments and reviews on the current implementation of `--security` argument.

File: lib/ansible/modules/packaging/os/yum.py [hunk 1]
```diff
@@ -136,6 +136,13 @@
     default: "/"
     aliases: []
 
+  security:
+    description:
+      - If set to C(yes), then installs all security updates.
```
Comment (afeld) [inline lib/ansible/modules/packaging/os/yum.py:141]: Might want to note that it's only applicable with `state: latest`.

Comment (Akasurde) [reply lib/ansible/modules/packaging/os/yum.py:141]: OK. I will update doc.

**Comment by abadger** [comment]
Added to devel for 2.4 release

**Pull request merged**